InTune/InTune/ViewControllers/EditProfController.swift

```diff
@@ -7,24 +7,154 @@
 //
 
 import UIKit
+import FirebaseAuth
+import Kingfisher
+/*
+ add password change + email change
+ add tag cv
+ */
 
 class EditProfController: UIViewController {
+    
+    @IBOutlet private var editImageView: DesignableImageView!
+    @IBOutlet private var usernameTextField: UITextField!
+    @IBOutlet private var bioTextView: UITextView!
+    
+    private lazy var imagePickerController: UIImagePickerController = {
+        let mediaTypes = UIImagePickerController.availableMediaTypes(for: .photoLibrary)
+        let pickerController = UIImagePickerController()
+        pickerController.mediaTypes = mediaTypes ?? ["kUTTypeImage"]
+        pickerController.delegate = self
+        return pickerController
+    }()
+    
+    public var selectedImage: UIImage? {
+        didSet{
+            editImageView.image = selectedImage
+        }
+    }
+    
+    private let storageService = StorageService()
+    let db = DatabaseService()
+    
+    var artist: Artist?
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        usernameTextField.delegate = self
+        updateUI()
+        getArtist()
+    }
+    
+    func getArtist(){
+        
+        guard let userID = Auth.auth().currentUser?.uid else {return}
+    
+        db.fetchArtist(userID: userID) { [weak self](result) in
+            switch result {
+            case.failure(let error):
+                print(error.localizedDescription)
+                
+            case.success(let artist1):
+                self?.artist = artist1
+                self?.usernameTextField.text = artist1.name
+                
+            }
+        }
+    }
+    
+    @IBAction func changeProfImagePressed(_ sender: UIButton) {
+        //photo library
+        imagePickerController.sourceType = .photoLibrary
+        present(imagePickerController, animated: true)
+    }
+    
+    func updateUI() {
+        guard let user = Auth.auth().currentUser else {
+            return
+        }
+        
+        if user.photoURL == nil  {
+                     editImageView.image = UIImage(systemName: "person.fill")
+                 } else {
+                   editImageView.kf.setImage(with: user.photoURL)
+        }
+        
+        usernameTextField.text = "\(user.displayName ?? "")"
 
-        view.backgroundColor = .systemBlue
+        
     }
     
+    func updateInfo() {
+        
 
-    /*
-    // MARK: - Navigation
+        guard let userName = usernameTextField.text, !userName.isEmpty, let selectedImage = editImageView.image, let bioText = bioTextView.text else {
+            showAlert(title: "Error editing", message: "Please check all fields")
+            return
+        }
+        
+        let resizedImage = UIImage.resizeImage(originalImage: selectedImage, rect: editImageView.bounds)
+        
+        print("orig \(selectedImage.size) not orig \(resizedImage)")
```

Comment:
there should not be any print statement in a production code. They do serve any purpose in production but they could leak some sensitive data if you are not careful.